src/kubeobjects/platform.go

```diff
@@ -15,22 +20,54 @@ const (
 	Openshift
 )
 
-func ResolvePlatformFromEnv() Platform {
-	switch os.Getenv(platformEnvName) {
-	case openshiftPlatformEnvValue:
-		return Openshift
-	case kubernetesPlatformEnvValue:
-		fallthrough
-	default:
-		return Kubernetes
+const SccGVR = "security.openshift.io/v1"
+
+type discoveryClientCreation interface {
+	getDiscoveryClient() (discovery.DiscoveryInterface, error)
+}
+
+type DiscoveryClientResolver struct{}
+
+func (p *DiscoveryClientResolver) getDiscoveryClient() (discovery.DiscoveryInterface, error) {
```

Comment:
THERE IS NOT EVEN THE LETTER `p` IN THE TYPE NAME 😂 
```suggestion
func (resolver *DiscoveryClientResolver) getDiscoveryClient() (discovery.DiscoveryInterface, error) {
```